core/domain/feature_gating_services.py

```diff
@@ -0,0 +1,177 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Services for platform parameters."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from constants import constants
+from core import features_registry
+from core.domain import platform_parameter_domain as param_domain
+from core.domain import platform_parameter_registry as registry
+
+
+ALL_FEATURES_LIST = (
+    features_registry.DEV_FEATURES_LIST +
+    features_registry.TEST_FEATURES_LIST +
+    features_registry.PROD_FEATURES_LIST
+)
+
+ALL_FEATURES_NAMES_SET = set(ALL_FEATURES_LIST)
+
+
+def create_evaluation_context_for_client(client_context_dict):
+    """Returns context instance for evaluation, using the information
+    provided by clients.
+
+    Args:
+        client_context_dict: dict. The client side context.
+
+    Returns:
+        EvaluationContext. The context for evaluation.
+    """
+    return param_domain.EvaluationContext.from_dict(
+        client_context_dict,
+        {
+            'server_mode': _get_running_mode()
+        }
+    )
+
+
+def get_all_feature_flag_setting_dicts():
+    """Returns dict representations of all feature flags.
+
+    Returns:
+        dict. The keys are the feature names and the values are the dict
+        mappings of all fields of the feature flags.
+    """
+    return dict(
+        (name, registry.Registry.get_platform_parameter(name).to_dict())
+        for name in ALL_FEATURES_LIST
+    )
+
+
+def get_all_feature_flag_values_for_context(context):
+    """Evaluates and returns the values for all feature flags.
+
+    Args:
+        context: EvaluationContext. The context used for evaluation.
+
+    Returns:
+        dict. The keys are the feature names and the values are boolean
+        results of corresponding flags.
+    """
+    return _get_feature_flag_values_for_context(ALL_FEATURES_LIST, context)
+
+
+def get_feature_flag_value(feature_name):
```

Comment:
I like the idea of simplifying the interface to ``is_feature_enabled``. I suggest keeping that pattern everywhere in Oppia (e.g. "is feature enabled' is always what we're checking in order to check the status of a feature).

I also suggest having a single public API function in this service to fulfill that role. Feel free to keep the evaluate method, but make it a local helper method if the intent is for service code to use ``is_feature_enabled`` to check the status of a feature.

Alternatively, we could have a slightly different API that involves interacting with the feature list directly. We could do something like:

```python
if platform_features.DUMMY_FEATURE.is_enabled:
    ...
```

Which would reduce the potential error of getting the name wrong. That obviously requires some reworking, but it might be a cleaner interface. I'd assume that ``is_enabled`` uses the evaluate function here, which also keeps the evaluation API of this service cleaner in that the two evaluation methods for client/server are consistent. What do you think?

Reply:
Added `is_feature_enabled` method, now `evaluate_feature_flag_value_for_server` is a local helper.

The alternative is interesting! I like how it works, but it seems like the typing system of python is not sufficient to provide feature name hint/check when we type something like `platform_features.DUMMY_`, unless we manually add the names one by one, instead of using loops or `__getattr__`. So, in my opinion, in python we still need to somehow manually type in the name of the features without pre-runtime hint, so then there's not much difference between this alternative and theh `is_feature_enabled(name)` solution -- developers still need to refer to the feature list to make sure they don't mistype anything..
In the frontend with typescript, I think it's possible to create an interface like this with type check, I will look into that in the following PR!

Reply:
Sounds good--thanks!